mllib/src/main/scala/org/apache/spark/mllib/linalg/Vectors.scala

```diff
@@ -712,6 +722,9 @@ class SparseVector @Since("1.0.0") (
     }
   }
 
+  @Since("1.6.0")
+  private[spark] def foreach(f: (Double) => Unit) = { indices.foreach { i: Int => f(values(i)) } }
```

Comment:
Does it? this is just iterating over the indices defined in the sparse vector. Hm, but i'm confused now.. the entries in `indices` are not indexes into the array `values`, but indexes into the sparse vector. This will fail if say indices=[3] and values=[0] when values[3] is dereferenced.